### PR TITLE
Removing old Preact Serve Command

### PR DESCRIPTION
**Pull request by RyanChristian4427**
Seeing as how `preact serve` no longer exists, I swapped it out with the same serve command that is used on the [default](https://github.com/preactjs-templates/default/blob/master/template/package.json) Preact template. This does add another dependency, but at least means the templates are similar. If you think having a "production-like" server isn't worth having around, I'm happy to remove it.

In order to satisfy `preact-render-spy`, I did add `@react/types` as recommended [here](https://github.com/preactjs/preact/issues/1036#issuecomment-491507106). It would be simple enough to make a correction by adding a `*d.ts` file containing the correct types, but that would be less extensible for future users, even if it does mean one less package is needed.

### CODE REVIEW

**Comment by lukeed** [comment]
Thanks! I'm not sure what the current plans are, but the last plan I heard was to use `sirv` since it includes a http/2 server, which is more production-like & takes the place of the old `simplehttp2server` binary.

The http/2 is available on `sirv-cli@next` right now

File: template/package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
         "preact-router": "^3.2.1"
     },
     "devDependencies": {
+        "@types/react": "^16.9.19",
```
Comment (ForsakenHarmony) [inline template/package.json]: this seems unrelated, also would prefer not having conflicts, I know the current situation isn't ideal

Comment (ForsakenHarmony) [reply template/package.json]: https://github.com/preactjs/preact/issues/2222

I hope this will provide a viable alternative

Comment (RyanChristian4427) [reply template/package.json]: I've replaced `@types/react` with a declaration file that exports the Preact JSX namespace, hopefully this is sufficient? Looks to be the simplest way to solve the issue while not adding too much extra to the template.

Comment (ForsakenHarmony) [reply template/package.json]: We can expand it later, or maybe compat can somehow include it
